File: java/is/tagomor/norikra/udf/AggregationSingleMock.java
package is.tagomor.norikra.udf;

import com.espertech.esper.epl.agg.aggregator.AggregationMethod;

// import java.util.Map;

public class AggregationSingleMock implements AggregationMethod {
  // private final static long = 16777216;
  private long counter;

  public AggregationSingleMock() {
    counter = 0;
  }

  public Class getValueType() {
    return Long.class;
  }

  public void enter(Object value) {
    if (value == null)
      return;

    Object[] params = (Object[]) value;
    for (Object o : params) {
    // for (int i = 0; params.length; i++) {
      //counter += params[i].toString().length;
      counter += o.toString().length();
    }
  }

  public void leave(Object value) {
    if (value == null)
      return;

    Object[] params = (Object[]) value;
    for (int i = 0; i < params.length; i++) {
      counter -= params[i].toString().length();
    }
  }

  public Object getValue() {
    return counter;
  }

  public void clear() {
    counter = 0;
  }
}

File: lib/esper_plugin/aggregation_single_mock.rb
require 'java'
java_package 'is.tagomor.norikra.udf'

java_import 'com.espertech.esper.epl.agg.aggregator.AggregationMethod'

class AggregationSingleMock
  java_implements 'AggregationMethod'

  def initialize
    @counter = 0
  end

  java_signature 'public Class getValueType()'
  def getValueType()
    java.lang.Long.java_class
  end

  java_signature 'public void enter(Object)'
  def enter(value)
    if value == nil
      return
    end

    @counter += value.to_s.length
  end

  java_signature 'public void leave(Object)'
  def leave(value)
    @counter -= value.to_s.length
  end

  java_signature 'public Object getValue()'
  def getValue()
    @counter
  end

  java_signature 'public void clear()'
  def clear()
    @counter = 0
  end
end
